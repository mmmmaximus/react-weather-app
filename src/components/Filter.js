import React from 'react';

const Filter = ({func}) => {
  return (
    <div className='filter'>
      find countries <input data-testid='Filter' onChange={func}/>
    </div>
  )
}

export default Filter
