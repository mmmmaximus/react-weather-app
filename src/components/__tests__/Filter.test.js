import { fireEvent, render, screen } from '@testing-library/react'
import Filter from '../Filter'
import renderer from 'react-test-renderer'

const filterFn = jest.fn()

test('Filter component should render', () => {
  render(<Filter func={filterFn} />)
  const FilterElement = screen.getByTestId('Filter')
  expect(FilterElement).toBeInTheDocument
})

test('should call function on input change', () => {
  render(<Filter func={filterFn} />)
  const FilterElement = screen.getByTestId('Filter')
  fireEvent.change(FilterElement, { target: { value: '1' } })
  expect(filterFn).toBeCalledTimes(1)
  fireEvent.change(FilterElement, { target: { value: '12' } })
  expect(filterFn).toBeCalledTimes(2)
})

test('it matches snapshot', () => {
  const tree = renderer.create(<Filter func={filterFn} />).toJSON()
  expect(tree).toMatchSnapshot()
})
