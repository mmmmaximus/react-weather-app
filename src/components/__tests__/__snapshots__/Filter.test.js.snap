// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div
  className="filter"
>
  find countries 
  <input
    data-testid="Filter"
    onChange={[MockFunction]}
  />
</div>
`;
